source/sigv4.c

```diff
@@ -436,6 +562,407 @@ static SigV4Status_t parseDate( const char * pDate,
 
 /*-----------------------------------------------------------*/
 
+/* Converts a hex character to its integer value */
+static char hexToInt( char pHex )
+{
+    uint8_t byte = pHex;
+
+    if( ( byte >= '0' ) && ( byte <= '9' ) )
+    {
+        byte = byte - '0';
+    }
+    else if( ( byte >= 'a' ) && ( byte <= 'f' ) )
+    {
+        byte = byte - 'a' + 10;
+    }
+    else if( ( byte >= 'A' ) && ( byte <= 'F' ) )
+    {
+        byte = byte - 'A' + 10;
+    }
+
+    return byte & 0xF;
+}
+
+/*-----------------------------------------------------------*/
+
+/* Hex digest of provided parameter string. */
+static void hexEncode( SigV4String_t * pInputStr,
+                       SigV4String_t * pHexOutput )
+{
+    static const unsigned char digitArr[] = "0123456789abcdef";
+    unsigned char * hex = pHexOutput->pData;
```

Comment:
Assertion should be done before accessing pHexOutput. This should also be asserted pHexOutput->pData